front/components/assistant/conversation/ActionValidationProvider.tsx

```diff
@@ -134,26 +144,37 @@ export function ActionValidationProvider({
 
   const handleSubmit = (approved: MCPValidationOutputType) => {
     void sendCurrentValidation(approved);
-    setIsProcessing(false);
+
     const foundItem = takeNextFromQueue();
-    if (!foundItem) {
+    if (foundItem) {
+      setCurrentValidation(foundItem);
+    } else {
+      // To avoid content flickering, we will clear out the current validation in onDialogAnimationEnd.
       setIsDialogOpen(false);
     }
   };
 
-  useEffect(() => {
-    if (currentValidation) {
-      setIsDialogOpen(true);
+  // To avoid content flickering, we will clear out the current validation when closing animation ends.
+  const onDialogAnimationEnd = () => {
```

Comment:
This will be triggered when dialog animation ends (can be opening and closing). I was wondering if `isDialogOpen` state will be up-to-date but actually it should be, because otherwise the dialog won't be closed